### ISSUE TITLE
CitrusFramework 2.7.6 - Param list support with same param name not available

### ISSUE THREAD

**Issue by rajivbansal2981** [citrusframework/citrus#583]
**Citrus Version**:CitrusFramework 2.7.6

**Expected behavior**
In citrus framework till 2.7.5, multiple queryparam could be sent in HttpClientRequestActionBuilder and same behaviour should exist with the new version 2.7.6

**Actual behavior**
In 2.7.6, not able to send the array params as queryparam input in the request in HttpClientRequestActionBuilder

**Test case sample**
Looked the source code found that in 2.7.6

```java
public HttpMessage queryParam(String name, String value) {
            if (!StringUtils.hasText(name)) {
                throw new CitrusRuntimeException("Invalid query param name -         must not be empty!");
    }

    this.queryParams.put(name, value);

    String queryParamString = queryParams.entrySet()
                                         .stream()
                                         .map(entry -> entry.getKey() + (entry.getValue() != null ? "=" + entry.getValue() : ""))
                                         .collect(Collectors.joining(","));

    header(HttpMessageHeaders.HTTP_QUERY_PARAMS, queryParamString);
    header(DynamicEndpointUriResolver.QUERY_PARAM_HEADER_NAME, queryParamString);

    return this;
} 
```

where as in release prior to 2.7.6 it was
```java
public HttpMessage queryParam(String name, String value) {
            if (!StringUtils.hasText(name)) {
                throw new CitrusRuntimeException("Invalid query param name - must not be empty!");
    }

    String queryParams;
    if (getHeader(HttpMessageHeaders.HTTP_QUERY_PARAMS) != null) {
        queryParams = getHeader(HttpMessageHeaders.HTTP_QUERY_PARAMS).toString();
        queryParams += "," + name + (StringUtils.hasText(value) ? "=" + value : "");
    } else {
        queryParams = name + (StringUtils.hasText(value) ? "=" + value : "");
    }

    header(HttpMessageHeaders.HTTP_QUERY_PARAMS, queryParams);
    header(DynamicEndpointUriResolver.QUERY_PARAM_HEADER_NAME, queryParams);

    return this;
}
```
As a result if i use `queryParam("paramName", "value1").queryParam("paramName", "value2")` then only paramName with "Value1" is sent not "value2".

May I ask why this feature is removed or is there any workaround available?

**Comment by svettwer** [comment]
Hi!

The change was made with #395. Unfortunately, I think this is a regression.

Based on the [Query Strings explanation on Wikipedia](https://en.wikipedia.org/w/index.php?title=Query_string&oldid=870843965)
>While there is no definitive standard, most web frameworks allow multiple values to be associated with a single field (e.g. field1=value1&field1=value2&field2=value3).[5][6]

In addition, the [rfc3986](https://tools.ietf.org/html/rfc3986#section-3.4) does not state anything that would forbid something like `field1=value1&field1=value2`.

Therefore I'm in favor to add this back again. Even if this is an important issue to fix in my opinion, this issue will not make it in v2.7.9. In addition, I currently see no workaround for this as the query parameters are stored in a map which wasn't the case before.

BR,
Sven

**Comment by svettwer** [comment]
closed with #634
